### PR TITLE
Generate trace ID in correct format (LocalLambda Server)

### PR DESCRIPTION
**Pull request by pokryfka**
Generate (X-Ray) Trace ID in correct format.

### Motivation:

Generate (X-Ray) Trace ID in correct format. Helpful when testing X-Ray tracing.

#### References

- [Generating trace IDs](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids)
- [Tracing header](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader)

### Modifications:

- different value of [Tracing header](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader) generated by `LocalLambda Server` with:
  - correctly formatted trace ID (root)
  - removed parent segment ID (which is optional)

### Result:

Example value of  `Lambda-Runtime-Trace-Id` header:

```
Root=1-5efd7f0b-da552a1eb226e4718fff8790;Sampled=1
```

before (root and parent values used to be signed integers):

```
"Root=25457;Parent=14820;Sampled=1"
```

### CODE REVIEW

**Comment by pokryfka** [comment]
Should not the value of `AmazonHeaders.traceID` be `X-Amzn-Trace-Id` instead of `Lambda-Runtime-Trace-Id`?

see [Tracing header](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader)

**Comment by pokryfka** [comment]
> @pokryfka This is awesome! I'm fine with this change but let's wait for @tomerd's opinion. Further, would you mind creating a test that verifies the format?

Sure, I can create a test.

In which case the function will need to be internal (its defined within private `LocalLambda` enum at the moment),
in which case.. should I move it to `Utils` or keep it in `Lambda+LocalServer`?
The function is generic though probably will only be used by the `LocalLambda:Server`.

@fabianfett

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Comment by tomerd** [comment]
thanks @pokryfka this is great. I think there are also places in the unit tests where we fake the traceID and would be nice to use a more correct one. 

wondering if we should rename the function to be generateXRayTraceID to clarify the intent. I also always like to add the "internal" modifier to make it super clear

**Comment by pokryfka** [comment]
@tomerd  Thank you for the feedback. Please see 2 questions below.

I could rename the function or/and create `XRay` namespace?

```swift
internal enum XRay {
    static func generateTraceID() -> String {
        // ...
    }
}
```

XRay related functions or types would end up there **if** needed or practical, for example we could also have:

```swift
internal enum XRay {
    static func generateTracingHeader(parent: String? = nil, sampled: Bool? = nil) -> String {
        // ...
    }
}
```

---

At the moment `LocalLambda:Server` is the only place where tracing header value is generated.

`MockLambdaServer` in `AWSLambdaRuntimeCoreTests` creates the tracing header using (correctly formatted) string literal; I dont think the actual values are used during tests; should they be generated?

```
                responseHeaders = [
                    (AmazonHeaders.requestID, requestId),
                    (AmazonHeaders.invokedFunctionARN, "arn:aws:lambda:us-east-1:123456789012:function:custom-runtime"),
                    (AmazonHeaders.traceID, "Root=1-5bef4de7-ad49b0e87f6ef6c87fc2e700;Parent=9a9197af755a6419;Sampled=1"),
                    (AmazonHeaders.deadline, String(deadline)),
                ]
```

**Comment by tomerd** [comment]
@parkera wdyt about adding it to `AmazonHeaders`?

**Comment by pokryfka** [comment]
@tomerd made both changes, please let me know if you have more comments

File: Sources/AWSLambdaRuntimeCore/Utils.swift [hunk 1]
```diff
@@ -104,3 +104,30 @@ extension String {
...
+    /// A `trace_id` consists of three numbers separated by hyphens.
+    /// For example, `1-58406520-a006649127e371903a2de979`. This includes:
+    /// - The version number, that is, 1.
+    /// - The time of the original request, in Unix epoch time, in **8 hexadecimal digits**.
+    /// For example, 10:00AM December 1st, 2016 PST in epoch time is `1480615200` seconds, or `58406520` in hexadecimal digits.
+    /// - A 96-bit identifier for the trace, globally unique, in **24 hexadecimal digits**.
+    /// # References
+    /// - [Generating trace IDs](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids)
+    /// - [Tracing header](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader)
+    @inlinable
```
Comment (tomerd) [inline Sources/AWSLambdaRuntimeCore/Utils.swift:120]: is @inlinable needed here?

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Comment by tomerd** [comment]
@pokryfka looks great, one question

File: Sources/AWSLambdaRuntimeCore/Utils.swift (hunk 1 above)
Comment (pokryfka) [reply Sources/AWSLambdaRuntimeCore/Utils.swift]: removed

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Comment by tomerd** [comment]
thank you @pokryfka <3

**Pull request merged**